src/main/java/uk/gov/hmcts/reform/userprofileapi/service/impl/UserProfileDeleter.java

```diff
@@ -0,0 +1,72 @@
+package uk.gov.hmcts.reform.userprofileapi.service.impl;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import uk.gov.hmcts.reform.userprofileapi.controller.response.UserProfilesDeletionResponse;
+import uk.gov.hmcts.reform.userprofileapi.domain.entities.UserProfile;
+import uk.gov.hmcts.reform.userprofileapi.exception.ResourceNotFoundException;
+import uk.gov.hmcts.reform.userprofileapi.repository.UserProfileRepository;
+import uk.gov.hmcts.reform.userprofileapi.resource.UserProfilesDeletionData;
+import uk.gov.hmcts.reform.userprofileapi.service.AuditService;
+import uk.gov.hmcts.reform.userprofileapi.service.ResourceDeleter;
+
+@Service
+@Slf4j
+public class UserProfileDeleter implements ResourceDeleter<UserProfilesDeletionData> {
+
+    @Autowired
+    private UserProfileRepository userProfileRepository;
+
+    @Autowired
+    private AuditService auditService;
+
+    @Override
+    @Transactional
+    public UserProfilesDeletionResponse delete(UserProfilesDeletionData profilesData) {
+
+        List<UserProfile> userProfiles = new ArrayList<UserProfile>();
+        Set<String> userIds = new HashSet<String>(profilesData.getUserIds());
+        userIds.forEach(userId -> userProfiles.add(validateUserStatus(userId)));
+
+        return deleteUserProfiles(userProfiles);
+    }
+
+    /**
+     * Either delete all the audit and userProfiles data from the data base or none.
+     *
+     */
+    private UserProfilesDeletionResponse deleteUserProfiles(List<UserProfile> userProfiles) {
+
+        userProfileRepository.deleteAll(userProfiles);
```

Comment:
if delete fails by some reason then will it give UserProfilesDeletionResponse in response ?

Reply:
it throws ErrorResponse from advice class .. UserProfilesDeletionResponse is for successful  case